test/xpu_api/utilities/meta/meta.unary/meta.unary.prop/is_constructible.pass.cpp

```diff
@@ -0,0 +1,243 @@
+            test_is_not_constructible<int, void() const>();
+            test_is_not_constructible<int&, void>();
+            test_is_not_constructible<int&, void()>();
+            test_is_not_constructible<int&, void() const>();
+            test_is_not_constructible<int&, void (&)()>();
+
+            test_is_not_constructible<void>();
+            test_is_not_constructible<const void>(); // LWG 2738
+            test_is_not_constructible<volatile void>();
+            test_is_not_constructible<const volatile void>();
+            test_is_not_constructible<int&>();
+            test_is_constructible<int, S>();
+            test_is_not_constructible<int&, S>();
+
+            test_is_constructible<void (&)(), void (&)()>();
+            test_is_constructible<void (&)(), void()>();
+            test_is_constructible<void(&&)(), void(&&)()>();
+            test_is_constructible<void(&&)(), void()>();
+            test_is_constructible<void(&&)(), void (&)()>();
+
+            test_is_constructible<int const&, int>();
+            test_is_constructible<int const&, int&&>();
+
+            test_is_constructible<void (&)(), void(&&)()>();
+
+            test_is_not_constructible<int&, int>();
+            test_is_not_constructible<int&, int const&>();
+            test_is_not_constructible<int&, int&&>();
+
+            test_is_constructible<int&&, int>();
+            test_is_constructible<int&&, int&&>();
+            test_is_not_constructible<int&&, int&>();
+            test_is_not_constructible<int&&, int const&&>();
+
+            test_is_constructible<Base, Derived>();
+            test_is_constructible<Base&, Derived&>();
+            test_is_not_constructible<Derived&, Base&>();
+            test_is_constructible<Base const&, Derived const&>();
+            test_is_not_constructible<Derived const&, Base const&>();
+            test_is_not_constructible<Derived const&, Base>();
+
+            test_is_constructible<Base&&, Derived>();
+            test_is_constructible<Base&&, Derived&&>();
+            test_is_not_constructible<Derived&&, Base&&>();
+            test_is_not_constructible<Derived&&, Base>();
+
+            // test that T must also be destructible
+            test_is_constructible<PrivateDtor&, PrivateDtor&>();
+            test_is_not_constructible<PrivateDtor, int>();
+
+            test_is_not_constructible<void() const, void() const>();
+            test_is_not_constructible<void() const, void*>();
+
+            test_is_constructible<int&, ImplicitTo<int&>>();
+            test_is_constructible<const int&, ImplicitTo<int&&>>();
+            test_is_constructible<int&&, ImplicitTo<int&&>>();
+            test_is_constructible<const int&, ImplicitTo<int>>();
+
+            test_is_not_constructible<B&&, B&>();
+            test_is_not_constructible<B&&, D&>();
+            test_is_constructible<B&&, ImplicitTo<D&&>>();
+            test_is_constructible<B&&, ImplicitTo<D&&>&>();
```

Comment:
Can we use `Base` and `Derived` instead of `B` and `D` here for consistency with other lines (e.g. 171-174?

Reply:
Fixed